### DISCUSSION TITLE
Error context.ɵgetModuleContext is not a function

### DISCUSSION THREAD

**Discussion by Sandeep-Malhotra** [Urigo/graphql-modules#2292] [General]
Error 
<img width="508" alt="image" src="https://user-images.githubusercontent.com/11954798/216888659-2c7abfa2-464f-455f-9436-e9b09dbc9afe.png">

code snippet
<img width="601" alt="image" src="https://user-images.githubusercontent.com/11954798/216888723-782de8df-f960-4661-992d-e2911d9e3fcd.png">

Full source code 
https://github.com/Sandeep-Malhotra/test-graphql

Kindly let me how to resolve it 

Thanks in advance

**Comment by dotansimha** [comment]
Hi @Sandeep-Malhotra 
Please follow the instructions in https://the-guild.dev/graphql/modules/docs/get-started#use-your-application (note the tab for Apollo Server). you need to use `application.createApolloExecutor()` and pass it to `ApolloServer`, in order to make it work.

**Reply by Sandeep-Malhotra** [reply]
Thanks for your response
Upon checking I found
Package @apollo/server is the latest one. But the example is using a deprecated package apollo-server.

Kindly advise how to use graphQL-modules with the new package apollo/server

**Reply by dotansimha** [reply]
We still don't have an example for v4 of Apollo-Server. It should be similar. 
@ardatan will update the examples soon.

**Reply by Sandeep-Malhotra** [reply]
I tired with graphQL-yoga I got same error

**Reply by dotansimha** [reply]
How did you set up Yoga with modules? Did you use the envelop plugin?

**Reply by Sandeep-Malhotra** [reply]
I was unable to do with envelop plugin ( useGraphQLModules )  . I tried below mentioned code
Code Snippets
```
import { createSchema, createYoga } from 'graphql-yoga'
import { createServer } from 'node:http'
import { createApplication  } from 'graphql-modules';
import { bookModule } from './modules/index.js';
import { useGraphQLModules } from '@envelop/graphql-modules'

const application = createApplication({
    modules:[bookModule]
});



const yoga = createYoga({
    schema: createSchema({
      typeDefs: application.typeDefs,
      resolvers: application.resolvers
    })
  })
  
  const server = createServer(yoga)
  
  server.listen(4000, () => {
    console.info('Server is running on http://localhost:4000/graphql')
  })
```

**Reply by dotansimha** [reply]
You should use `useGraphQLModules` because that's the plugins that makes sure to extend the context with the required properties, like `ɵgetModuleContext`

**Comment by MatamandoKalilani** [comment]
Check out this link - https://github.com/Urigo/graphql-modules/issues/2270#issuecomment-1365000062

I think I was facing a similar problem, and it seemed to solve the issue. 

Hope this helps.
